pkg/bpf/prog.go

```diff
@@ -67,3 +74,72 @@ func (t ProgType) String() string {
 
 	return "Unknown"
 }
+
+type attrProg struct {
+	startID   uint32
+	progID    uint32
+	nextID    uint32
+	openFlags uint32
+}
+
+// ProgInfo holds values from the upstram struct bpf_prog_info.
+// From: https://github.com/torvalds/linux/blob/master/include/uapi/linux/bpf.h#L1006
+type ProgInfo struct {
+	ProgType     ProgType
+	ID           uint32
+	MapIDs       uint64
+	CreatedByUID uint32
+	LoadTime     uint64 /* ns since boottime */
+	Name         string
+}
+
+type attrObjInfo struct {
+	bpfFD   uint32
+	infoLen uint32
+	info    uint64
+}
+
+// GetNextID takes a program ID and returns the next program ID.
+func GetNextID(start, next unsafe.Pointer) error {
```

Comment:
What are the actual types that users should be using here? 32-bit integers for the program ids? Perhaps we could change this to something like `func GetNextID(start, next *uint32) error {`. Then the caller is guaranteed to provide the right parameter type.